op-challenger/game/fault/types/actions.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+import "github.com/ethereum/go-ethereum/common"
+
+type ActionType string
+
+func (a ActionType) String() string {
+	return string(a)
+}
+
+const (
+	ActionTypeMove ActionType = "move"
+	ActionTypeStep ActionType = "step"
+)
+
+type Action struct {
+	Type      ActionType
+	ParentIdx int
+	IsAttack  bool
```

Comment:
Since we're refactoring the `Action` here, what do you think about remove the `IsAttack` field in `Action` and making `ActionType`s more verbose to include whether the action is an attack or defense?

For example
```golang
const (
	AttackMoveAction ActionType = "attack_move"
        DefendMoveAction ActionType = "defend_move"
	AttackStepAction ActionType = "attack_step"
        DefendStepAction ActionType = "defend_step"
)
```

Reply:
I actually started that way and found that it was really painful. You wind up doing a lot of `if Type == AttackMoveAction || Type == DefendMoveAction` type stuff.